packages/core/src/tracing/instrumentation/cloud/azure/blob.js

```diff
@@ -0,0 +1,167 @@
+/*
+ * (c) Copyright IBM Corp. 2021
+ * (c) Copyright Instana Inc. and contributors 2018
+ */
+
+'use strict';
+
+const shimmer = require('../../../shimmer');
+
+const requireHook = require('../../../../util/requireHook');
+const tracingUtil = require('../../../tracingUtil');
+const constants = require('../../../constants');
+const cls = require('../../../cls');
+
+let isActive = false;
+
+exports.spanName = 'azstorage';
+
+exports.init = function init() {
+    requireHook.onModuleLoad('@azure/storage-blob', instrumentBlob);
+};
+
+function instrumentBlob(blob) {
+    instrumentBlockBlob(blob.BlockBlobClient);
+    instrumentContainer(blob.BlobClient);
+}
+function instrumentContainer(blobClient) {
+    shimmer.wrap(blobClient.prototype, 'delete', shimDelete);
+    shimmer.wrap(blobClient.prototype, 'download', shimDownload);
+}
+function instrumentBlockBlob(blockBlobClient) {
+    shimmer.wrap(blockBlobClient.prototype, 'upload', shimUpload);
+    shimmer.wrap(blockBlobClient.prototype, 'stageBlock', shimUpload);
```

Comment:
When data length is more than that of specified as maxSingleShotSize , stageBlock will be called to upload blocks.
it is tested in [testcase](https://github.com/instana/nodejs/blob/fe35e3d476117c49f24725e3728c7a56a9305895/packages/collector/test/tracing/cloud/azure/blob/app.js#L254)

Reply:
Okay, but if I comment out:

// shimmer.wrap(blockBlobClient.prototype, 'stageBlock', shimUpload);

all tests still pass. So something is wrong?

Reply:
Testsa re passing because, the span parameters for **stage block  upload** doesnt really differ from that of **upload**.  The value of op in span data is **upload** in both case, since the opertaion is ultimately blob upload for the user. Any change required in span data that will be useful to end user?

Reply:
If the `stageBlock` function would have been called, the test should fail if I comment out this line?
I do not understand your reasoning.

Reply:
My mistake. The testfile that got pushed to repo doesnt invoke the stageblock (file size is lesser in KB ). fixed that in testcase by adjusting maxSingleShotSize.